streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -253,6 +260,23 @@ public void onRestoreEnd(TopicPartition topicPartition, String storeName, long t
         }
         consumer.assign(partitionsByTopic.values());
 
+
+        final Sensor skippedRecordsSensor = metrics.sensor("skipped-records");
+        final MetricName rateMetricName = metrics.metricName(
+            "skipped-records-rate",
+            "stream-metrics",
+            "The average per-second number of skipped records",
+            Collections.singletonMap("client-id", "topology-test-driver")
+        );
+        final MetricName totalMetricName = metrics.metricName(
+            "skipped-records-total",
+            "stream-metrics",
+            "The total number of skipped records",
+            Collections.singletonMap("client-id", "topology-test-driver")
+        );
+
+        skippedRecordsSensor.add(new Meter(new Sum(), rateMetricName, totalMetricName));
```

Comment:
Adding this implies, that we have to maintain the same code twice. Should we extract this into some internal method that we can call here to avoid code duplication?

What about other thread-level metrics?

Reply:
I've been mulling over the same thing, and that was part of what I was trying to achieve with my experiment before. I think I have a better solution now, so maybe you can take another look after my next update and see what you think.

Reply:
Will do, after you pushed an update :)